src/main/CommandHandler.cpp

```diff
@@ -768,6 +770,107 @@ CommandHandler::scpInfo(std::string const& params, std::string& retStr)
+    root["RUN_STANDALONE"] = config.RUN_STANDALONE;
+    root["MANUAL_CLOSE"] = config.MANUAL_CLOSE;
+    root["CATCHUP_COMPLETE"] = config.CATCHUP_COMPLETE;
+    root["CATCHUP_RECENT"] = config.CATCHUP_RECENT;
+    root["MAINTENANCE_ON_STARTUP"] = config.MAINTENANCE_ON_STARTUP;
+    root["ARTIFICIALLY_GENERATE_LOAD_FOR_TESTING"] = config.ARTIFICIALLY_GENERATE_LOAD_FOR_TESTING;
+    root["ARTIFICIALLY_ACCELERATE_TIME_FOR_TESTING"] = config.ARTIFICIALLY_ACCELERATE_TIME_FOR_TESTING;
+    root["ARTIFICIALLY_SET_CLOSE_TIME_FOR_TESTING"] = config.ARTIFICIALLY_SET_CLOSE_TIME_FOR_TESTING;
+    root["ARTIFICIALLY_PESSIMIZE_MERGES_FOR_TESTING"] = config.ARTIFICIALLY_PESSIMIZE_MERGES_FOR_TESTING;
+    root["ALLOW_LOCALHOST_FOR_TESTING"] = config.ALLOW_LOCALHOST_FOR_TESTING;
+    root["FAILURE_SAFETY"] = config.FAILURE_SAFETY;
+    root["UNSAFE_QUORUM"] = config.UNSAFE_QUORUM;
+    root["LEDGER_PROTOCOL_VERSION"] = config.LEDGER_PROTOCOL_VERSION;
+    if (config.PREFERRED_UPGRADE_DATETIME) {
+        root["PREFERRED_UPGRADE_DATETIME"] = VirtualClock::tmToISOString(*config.PREFERRED_UPGRADE_DATETIME);
+    }
+    root["OVERLAY_PROTOCOL_MIN_VERSION"] = config.OVERLAY_PROTOCOL_MIN_VERSION;
+    root["OVERLAY_PROTOCOL_VERSION"] = config.OVERLAY_PROTOCOL_VERSION;
+    root["VERSION_STR"] = config.VERSION_STR;
+    root["LOG_FILE_PATH"] = config.LOG_FILE_PATH;
+    root["BUCKET_DIR_PATH"] = config.BUCKET_DIR_PATH;
+    root["DESIRED_BASE_FEE"] = config.DESIRED_BASE_FEE;
+    root["DESIRED_BASE_RESERVE"] = config.DESIRED_BASE_RESERVE;
+    root["DESIRED_MAX_TX_PER_LEDGER"] = config.DESIRED_MAX_TX_PER_LEDGER;
+    root["HTTP_PORT"] = config.HTTP_PORT;
+    root["PUBLIC_HTTP_PORT"] = config.PUBLIC_HTTP_PORT;
+    root["HTTP_MAX_CLIENT"] = config.HTTP_MAX_CLIENT;
+    std::string NETWORK_PASSPHRASE; // identifier for the network
+    root["PEER_PORT"] = config.PEER_PORT;
+    root["TARGET_PEER_CONNECTIONS"] = config.TARGET_PEER_CONNECTIONS;
+    root["MAX_PEER_CONNECTIONS"] = config.MAX_PEER_CONNECTIONS;
+    addVector("PREFERRED_PEERS", config.PREFERRED_PEERS);
+    addVector("KNOWN_PEERS", config.KNOWN_PEERS);
+    addVector("PREFERRED_PEER_KEYS", config.PREFERRED_PEER_KEYS);
+    root["PREFERRED_PEERS_ONLY"] = config.PREFERRED_PEERS_ONLY;
+    root["MINIMUM_IDLE_PERCENT"] = config.MINIMUM_IDLE_PERCENT;
+    root["MAX_CONCURRENT_SUBPROCESSES"] = static_cast<uint32_t>(config.MAX_CONCURRENT_SUBPROCESSES);
+    root["NODE_IS_VALIDATOR"] = config.NODE_IS_VALIDATOR;
+    root["INVARIANT_CHECK_BALANCE"] = config.INVARIANT_CHECK_BALANCE;
+    root["INVARIANT_CHECK_ACCOUNT_SUBENTRY_COUNT"] = config.INVARIANT_CHECK_ACCOUNT_SUBENTRY_COUNT;
+    root["INVARIANT_CHECK_CACHE_CONSISTENT_WITH_DATABASE"] = config.INVARIANT_CHECK_CACHE_CONSISTENT_WITH_DATABASE;
+    root["QUORUM_SET"] = makeQuorumSetNode(config.QUORUM_SET);
+    
+    Json::Value validatorNames;
+    for (auto const& kv : config.VALIDATOR_NAMES) {
+        validatorNames.append(kv.second);
```

Comment:
could you make this into an json associative array matching this C++ array?

Reply:
Done.